src/test/Fake.DotNet.Cli.IntegrationTests/TemplateTests.fs

```diff
@@ -0,0 +1,91 @@
+﻿module Fake.DotNet.Cli.IntegrationTests.TemplateTests
+
+open Expecto
+open System
+open System.IO
+
+open Fake.Core
+open Fake.DotNet
+open Fake.IO
+open Fake.IO.Globbing
+
+let templateProj = "fake-template.fsproj"
+let templatePackageName = "fake-template"
+let templateName = "fake"
+
+//TODO: add DotNetCli helpers for the `new` command
+
+let uninstallTemplate () =
+    DotNet.exec id "new" (sprintf "-u %s" templatePackageName)
+
+let installTemplateFrom pathToNupkg =
+    DotNet.exec id "new" (sprintf "-i %s" pathToNupkg)
+
+type BootstrapKind =
+| Tool
+| Project
+| None
+with override x.ToString () = match x with | Tool -> "tool" | Project -> "project" | None -> "none"
+
+let shouldSucceed message (r: ProcessResult) =
+    Expect.isTrue r.OK (sprintf "%s. Results:\n:%A" message r)
+
+let runTemplate rootDir kind = async {
+    Directory.ensure rootDir
+    let dotnet = DotNet.Options.Create().DotNetCliPath
+    let proc = 
+        Process.getProc (fun p ->
+            p.WithWorkingDirectory(rootDir)
+             .WithFileName(dotnet)
+             .WithArguments(sprintf "new %s --bootstrap %s" templateName (string kind))
+             .WithRedirectStandardInput(true)
+             .WithRedirectStandardOutput(true)
+             .WithRedirectStandardError(true)
+        )
+    
+    proc.Start () |> ignore
+    do! Async.Sleep 2000
+    proc.StandardInput.WriteLine("Y")
+    proc.WaitForExit ()
+    Expect.equal proc.ExitCode 0 "should have run the template successfully"
+}
+
+let invokeScript dir scriptName =
+    let fullScriptPath = Path.Combine(dir, scriptName)
+    Process.execWithResult 
+        (fun x -> 
+            x.WithWorkingDirectory(dir)
+             .WithFileName(fullScriptPath)
+             .WithArguments "--help" ) (System.TimeSpan.FromSeconds 60.)
+    |> shouldSucceed "should invoke the script file"
```

Comment:
I'd prefer a test which actually checks if `./fake build` works (ie all dependencies still work)
Also can we add a failing test (ie that the scripts correctly return nonzero exit-codes on failure). For example by executing `./fake.sh run nonexisting.fsx`.

Reply:
Can do